app/products/calls/connection/websocket_client.ts

```diff
@@ -104,12 +141,35 @@ export default class WebSocketClient extends EventEmitter {
 
     close() {
         if (this.ws) {
+            this.closed = true;
             this.ws.close();
             this.ws = null;
+            this.seqNo = 1;
+            this.serverSeqNo = 0;
+            this.connID = '';
+            this.originalConnID = '';
+        } else {
+            this.emit('close');
+
+            const now = Date.now();
+            if (this.lastDisconnect === 0) {
+                this.lastDisconnect = now;
+            }
+
+            if ((now - this.lastDisconnect) >= wsReconnectionTimeout) {
+                this.closed = true;
+                this.emit('error', wsReconnectionTimeoutErr);
+                return;
+            }
+
+            setTimeout(() => {
+                if (!this.ws && !this.closed) {
+                    this.init(true);
```

Comment:
if I understand correctly, this WS will always attempt to stay connected, do we have a way to manually disconnect and if that is the case, do we want to prevent this from attempting the reconnection?

Reply:
I believe that's coming soon with a change to the client leave event.